### ISSUE TITLE
Auto update imports on file move is extremelly slow

### ISSUE THREAD

**Issue by danielo515** [microsoft/vscode#102750]
Issue Type: <b>Bug</b>

Hello.
Auto update imports when I rename or move a file is so slow that I'm able to edit them manually and half a minute after I get an error on the import because VSCode has tried to update the import after I already modified it, ending with a weird string that contains my rename and the one that VSCode did.
So, if I have a file named `feature.js` and an import like `import feature from './feature'`, I can rename it to featureModal.js, update the import and after 30 seconds what I get is `import feature from ./featureModalModal`

This happens on a workspace where I have two folders,both javascript projects

VS Code version: Code 1.47.1 (485c41f9460bdb830c4da12c102daff275415b53, 2020-07-13T22:43:02.577Z)
OS version: Darwin x64 19.5.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz (12 x 2900)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 3, 3|
|Memory (System)|32.00GB (0.46GB free)|
|Process Argv|-psn_0_626841|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (83)</summary>

Extension|Author (truncated)|Version
---|---|---
yaml2json|ahe|0.0.3
project-manager|ale|11.1.0
auto-add-brackets|ali|0.11.0
Handlebars|and|0.4.1
jest-snippets|and|1.8.0
vscode-django|bat|0.20.0
better-toml|bun|0.3.2
npm-intellisense|chr|1.3.0
keymapceditor-vsc|Cia|0.7.6
vscode-devtools-for-chrome|cod|0.0.7
bracket-pair-colorizer|Coe|1.0.61
vscode-svgviewer|css|2.0.0
danielo-node-snippets|dan|1.8.1
danielo-vsc-tools|dan|1.0.1
vscode-eslint|dba|2.1.8
javascript-ejs-support|Dig|1.3.1
python-extension-pack|don|1.6.0
jest-test-language-support|dpl|1.30.2
es7-react-js-snippets|dsz|2.8.2
gitlens|eam|10.2.2
EditorConfig|Edi|0.15.1
prettier-vscode|esb|5.1.3
vimL|fal|0.0.3
flow-for-vscode|flo|1.5.0
auto-rename-tag|for|0.1.4
reasonml|fre|1.0.38
vscode-flow-ide|gca|1.2.0
monkey-c|ghi|0.4.0
lintlens|ghm|3.0.0
vscode-pull-request-github|Git|0.18.0
go|gol|0.15.1
githd|hui|2.2.4
yuml|Jai|3.5.1
reason-vscode|jar|1.7.10
vscode-peacock|joh|3.7.2
styled-components-snippets|jon|0.6.0
open-in-vim|jon|0.6.0
vscode-styled-components|jpo|0.0.29
nim|kos|0.6.6
MagicPython|mag|1.1.0
vscode-wrap-console-log|mid|1.7.2
dotenv|mik|1.0.1
vscode-docker|ms-|1.3.1
remote-containers|ms-|0.128.0
remote-ssh|ms-|0.51.0
remote-ssh-edit|ms-|0.51.0
remote-ssh-explorer|ms-|0.51.0
remote-wsl|ms-|0.44.4
vscode-remote-extensionpack|ms-|0.20.0
cpptools|ms-|0.29.0
vscode-typescript-next|ms-|4.0.20200715
debugger-for-chrome|msj|4.12.9
vscode-purty|mva|0.4.1
color-highlight|nau|2.3.0
language-purescript|nwo|0.2.4
indent-rainbow|ode|7.4.0
vscode-react-refactor|pla|1.0.5
vscode-graphql|Pri|0.2.14
vscode-yaml|red|0.9.1
partial-diff|ryu|1.4.1
vscode-javascript-booster|sbu|14.0.1
bash-beautify|sha|0.1.1
code-settings-sync|Sha|3.4.3
tiddlywiki5|sin|0.0.2
vscode-fileutils|sle|3.0.1
addDocComments|ste|0.0.8
code-spell-checker|str|1.9.0
control-snippets|svi|1.9.1
json-schema-validator|tbe|0.1.0
open-spec-file|the|0.0.6
highlight-matching-tag|vin|0.9.9
vscodeintellicode|Vis|1.2.9
template-strings|vit|0.0.6
vscode-icons|vsc|10.2.0
vim|vsc|1.15.0
vscode-wakatime|Wak|4.0.0
vscode-file-extra|wil|3.0.0
vscode-import-cost|wix|2.12.0
change-case|wma|1.0.0
disable-eslint-rule|Woo|0.0.2
markdown-all-in-one|yzh|3.1.0
vscode-open-in-github|ziy|1.3.6
es6-string-javascript|zjc|1.0.1

</details>

**Comment by gjsjohnmurray** [comment]
Does it also happen if you reload with all extensions disabled? Maybe one or more of your 83 extensions is slowing things down.

**Comment by danielo515** [comment]
Yeah,that can be it. In fact I'm pretty sure it has to do with the VIM extension, which generally slows down the entire thing.
In any case, I have many of the extensions disabled on a per-workspace basis. I'll try anyway

**Comment by danielo515** [comment]
I tried with extensions disabled and it still happens, I was able to add the new name to the file that was importing the other file before VSCode had a chance

**Comment by gjsjohnmurray** [comment]
How many files are open when you do the rename?

I'm asking these questions in the hope of giving @mjbvz a head start when he gets to this.

**Comment by danielo515** [comment]
Maybe 8 - 12, I don't like having many editors open. It happens on a workspace that consists in two Javascript directories which are inside a bigger golang project (which I don't open with vscode entire because it's too big)

**Comment by mjbvz** [comment]
Can you share an example project that demonstrates this issue?

Also, are you using a workspace version of TypeScript? If so, which version?

**Comment by danielo515** [comment]
Do you mean if I have typescript installed on my workspace? If that is the question the answer is no, I'm using flow.

Isn't any other way to give you insights? Like logs or so. I'm pretty sure this only happens on big repos, which is not easy to reproduce

**Comment by mjbvz** [comment]
Not sure if this is the issue but we don't support flow. If you normal JS files are using flow conventions, this may break VS Code's built-in JS intellisense

To collect the logs:

1. Set `"typescript.tsserver.log": "verbose",`
1. Restart VS Code and reproduce the problem
1. In VS Code, run the `TypeScript: Open TS Server log` command
1. This should open a large log file called `tsserver.log` 


 > **⚠️Warning:** The TypeScript log may include information from your workspace, including file paths and source code. If you have any concerns about posting this publicly on Github, just let me know and we can arrange something else. On our side, we only use these logs to investigate issues like this

**Comment by danielo515** [comment]
Yes, we use flow for all the js. To be honest, the behavior of VSCode is a bit erratic in our repository. Sometimes the intellisense works, sometimes doesn't and this has started happening recently. I'll provide you the logs asap, thanks

**Comment by mjbvz** [comment]
Thanks, but just keep in mind that VS Code does not support flow (even if most things appear to work ok). Try using the Flow extension and disabling VS Code's built-in JS extension

**Comment by danielo515** [comment]
> Thanks, but just keep in mind that VS Code does not support flow (even if most things appear to work ok). Try using the Flow extension and disabling VS Code's built-in JS extension

I have the JScheck of typescript disabled and I'm using the flow extension.
By the way, how I'm supposed to restart vscode? I just executed the reload window command, but it seems that the debug log is still on info

**Comment by danielo515** [comment]
[tsserver.log](https://github.com/microsoft/vscode/files/5026668/tsserver.log)
In any case, here is the tsserver log
Let me know when I can delete it. It does not contain any sensitive information,but I don't like to have it lying around for no reason

**Comment by mjbvz** [comment]
Hm, I only see a single (relatively fast) call to `GetEditsForFileRename` in the logs. That's  what we use to update imports.

Could the import updating be coming from the flow extension instead? 

Also, as I noted, we also just do not support flow. Disabling validation will suppress the parsing errors but you probably want to disable the js/ts extension entirely.

**Comment by danielo515** [comment]
> Also, as I noted, we also just do not support flow. Disabling validation will suppress the parsing errors but you probably want to disable the js/ts extension entirely.

Then I will not get any intellisense and go-to-file at all? Or will the flow extension take care of that?
By the way, not sure if I mentioned but this is a pretty big repository. Right now I'm using it by opening just two subfolders as part of a workspace, can that have any impact?

> Could the import updating be coming from the flow extension instead?

Is there any way I can check if that is the case?

I can provide a screen recording if you want to take a look at it live.

**Comment by mjbvz** [comment]
It's actually worse: because Flow uses normal `.js` files, VS Code's JS/TS IntelliSense will treat Flow code as normal JavaScript. While some things will appear to work, Flow specific syntax may cause the IntelliSense engine to have major problems (including possibly breaking your code on operations like format or organize imports)

Again, if you are using Flow, disable the built-in JS/TS extension and use that extension instead. If it does not support the auto rename feature, file a feature request against the extension

**Comment by danielo515** [comment]
Ok, I'll try that, thanks @mjbvz
